common/src/main/java/com/wynntils/features/user/players/PreventTradesDuelsFeature.java

```diff
@@ -14,25 +16,39 @@
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.Event;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 @FeatureInfo(category = FeatureCategory.PLAYERS)
 public class PreventTradesDuelsFeature extends UserFeature {
+    private static final int FIGHTING_CUTOFF = 30; // 30 seconds
+
+    @Config
+    public boolean onlyWhileFighting = true;
 
     @SubscribeEvent
     public void onPlayerRightClick(PlayerInteractEvent.Interact event) {
-        if (!shouldBlockClick(event.getPlayer(), event.getItemStack(), event.getTarget())) return;
-
-        // stops interact packet from going out
-        event.setCanceled(true);
+        handlePlayerClick(event, event.getPlayer(), event.getItemStack(), event.getTarget());
     }
 
     @SubscribeEvent
     public void onPlayerLeftClick(PlayerAttackEvent event) {
-        if (!shouldBlockClick(event.getPlayer(), event.getPlayer().getMainHandItem(), event.getTarget())) return;
+        handlePlayerClick(event, event.getPlayer(), event.getPlayer().getMainHandItem(), event.getTarget());
+    }
 
-        // stops attack packet from going out
+    private void handlePlayerClick(Event event, Player player, ItemStack itemStack, Entity target) {
+        int timeSinceLastFight = (int) ((System.currentTimeMillis() - Models.CombatXp.getLastXpGainTimestamp()) / 1000);
```

Comment:
This is not fully what the PR title means. There are many cases where you don't gain any XP, yet you are still fighting. Like raid bosses. Those are places you REALLY don't want to trade, but you still do a lot of shift-clicking to trigger powder specials. 

Preferably we should implement the DPS tracker from legacy (and add `area_damage_per_second` function, as it is one the most requested one at the moment.) and use it here. Until that happens, can we add a FIXME here (and somehow rename the PR to reflect this)?

Reply:
Yes, you are correct. I used "time since last combat xp" as a proxy to mean "is fighting". Most of the cases when you fight a boss there will be minion mobs that are killed in the process, so I think this will work fine in most cases. But I did consider the possibility of a single, powerful boss as a way that this could fail. 

Note that the i18n description clarifies that "is fighting" counts as being less than 30 seconds since the last killed mob. I know that RTFM is a bad excuse for creating lousy programs, but at least it is documented.

I was toying with the possibility of tracking spells and melee attacks, but it was just too hard. Spells could be teleport spells, melee attacks differs between archer and others, and maybe you are just swinging your fist in the air to say hi, etc. 

If we could track damage dealt it would be an improvement, yes. How did we do that in legacy? Count the heart popup AS texts?

Reply:
> If we could track damage dealt it would be an improvement, yes. How did we do that in legacy? Count the heart popup AS texts?

AFAIK, we tracked the pop-up "labels". You would have to check to make sure though.

The second, more important question is what we should do with the PR. Imo, accept it. Even if you want to do damage tracking, there is no need to be in this PR, and it can be a direct follow up.